### PR TITLE
Adds Solr metrics gathering to jmx-metrics

### PR DESCRIPTION
**Pull request by StefanKurek**
**Description:**

* Adds Solr metric support

**Testing:**

* Adds Integration Tests for Solr metric gathering

**Documentation:**

* Adds documentation for Solr metrics

### CODE REVIEW

File: jmx-metrics/src/main/resources/target-systems/solr.groovy [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+   "type" : { mbean -> mbean.name().getKeyProperty("category") },
+   "handler" : { mbean -> mbean.name().getKeyProperty("scope") }],
+  "Count", otel.&longCounterCallback)
+
+def beanSolrCoreQueryResultsCache = otel.mbean("solr:dom1=core,dom2=*,category=CACHE,scope=*,name=queryResultCache")
+otel.instrument(beanSolrCoreQueryResultsCache, "solr.cache.eviction.count", "The number of evictions from a cache.", "evictions",
+  ["core" : { mbean -> mbean.name().getKeyProperty("dom2") },
+   "cache" : { mbean -> mbean.name().getKeyProperty("scope") }],
+  "cumulative_evictions", otel.&longCounterCallback)
+otel.instrument(beanSolrCoreQueryResultsCache, "solr.cache.hit.count", "The number of hits for a cache.", "hits",
```
Comment (dehaansa) [inline jmx-metrics/src/main/resources/target-systems/solr.groovy:66]: Feels like hits/inserts could be combined to one metric of "solr.cache.lookups.count" and some kind of label of "hit/miss", and lookups would be unnecessary.

File: jmx-metrics/docs/target-systems/solr.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+* Name: `solr.request.error.count`
+* Description: The number of queries resulting in an error.
+* Unit: `queries`
+* Labels: `core`, `type`, `handler`
+* Instrument Type: ObservableLongCounter
+
+
+* Name: `solr.request.error.count`
```
[the remark this answers is not in the document]
Comment (StefanKurek) [reply jmx-metrics/docs/target-systems/solr.md:45]: yes! good catch. i made this change but on a different branch. looks like i forgot to add it here.

File: jmx-metrics/src/main/resources/target-systems/solr.groovy (hunk 1 above)
Comment (StefanKurek) [reply jmx-metrics/src/main/resources/target-systems/solr.groovy:66]: @dehaansa miss actually doesn't appear to be a metric that we can directly get, so i think it's ok as is. i see your reasoning though

Comment (dehaansa) [reply jmx-metrics/src/main/resources/target-systems/solr.groovy:66]: Sounds good.

**Review by dehaansa — APPROVED** [review]

**Comment by StefanKurek** [comment]
@rmfitzpatrick Hello. Could you please take a look at this when you get a moment. Thanks!

File: jmx-metrics/docs/target-systems/solr.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Solr Metrics
+
+The JMX Metric Gatherer provides built in Solr metric gathering capabilities.
```
Comment (rmfitzpatrick) [inline jmx-metrics/docs/target-systems/solr.md:3]: Do you know of a jmx monitoring reference in the solr docs that can be linked to?

File: jmx-metrics/docs/target-systems/solr.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+* Name: `solr.request.count`
+* Description: The number of queries made.
+* Unit: `{queries}`
+* Labels: `core`, `type`, `handler`
+* Instrument Type: ObservableLongCounter
+
+
+* Name: `solr.request.time.average`
+* Description: The average time of a query. This average is based on the histogram configuration
```
Comment (rmfitzpatrick) [inline jmx-metrics/docs/target-systems/solr.md:31]: nit - this seems like a fairly long description, but has useful information. Maybe something like `The average time of a query, based on Solr's histogram configuration.`? I'm not sure if "EDR" is an accepted abbreviation in dropwizard but I see a few references when searching around.

File: jmx-metrics/README.md [hunk 5]
```diff
@@ -68,12 +68,13 @@ mutually exclusive with `otel.jmx.groovy.script`. The currently supported target
 
 | `otel.jmx.target.system` |
 | ------------------------ |
-| [`jvm`](./docs/target-systems/jvm.md) |
 | [`activemq`](./docs/target-systems/activemq.md)|
 | [`cassandra`](./docs/target-systems/cassandra.md) |
+| [`jvm`](./docs/target-systems/jvm.md) |
```
[the remark this answers is not in the document]
Comment (StefanKurek) [reply jmx-metrics/README.md:73]: @rmfitzpatrick Just for clarification I won't plan on changing unless you think it's necessary.

File: jmx-metrics/docs/target-systems/solr.md (hunk 4 above)
Comment (StefanKurek) [reply jmx-metrics/docs/target-systems/solr.md]: @rmfitzpatrick I have made the change you suggested. I like your shortening.

**Comment by StefanKurek** [comment]
@rmfitzpatrick please take a look again when you have the chance

File: jmx-metrics/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply jmx-metrics/README.md:73]: Let's keep it first for now if you don't mind. When I have a moment I'm going to try to polish some docs and will separate it.

File: jmx-metrics/docs/target-systems/solr.md (hunk 3 above)
Comment (rmfitzpatrick) [reply jmx-metrics/docs/target-systems/solr.md:3]: I'm not versed in solr but https://solr.apache.org/guide/6_6/using-jmx-with-solr.html seems like a fine launching point to show it's a "blessed" monitoring interface

Comment (StefanKurek) [reply jmx-metrics/docs/target-systems/solr.md:3]: @rmfitzpatrick I can throw in a link to it.

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**